src/counts.jl

```diff
@@ -388,7 +394,15 @@ of occurrences.
 """
 countmap(x::AbstractArray{T}; alg = :auto) where {T} = addcounts!(Dict{T,Int}(), x; alg = alg)
 countmap(x::AbstractArray{T}, wv::AbstractVector{W}) where {T,W<:Real} = addcounts!(Dict{T,W}(), x, wv)
-
+# fall-back for iterator `x`
+function countmap(x)
```

Comment:
The problem with this approach is that if the user calls `addcounts!` directly (which is part of the public API), the fast method won't be used. I think the solution is to have `addcounts!(cm::Dict, x; alg = :ignored) = _addcounts!(eltype(x), cm, x)` and turn all existing `addcounts!` methods into `_addcounts!`, dispatching on the eltype.

Reply:
the problem is dispatch doesn't work on `alg=:ignored` keyword. idk, maybe for a different issue?

Reply:
I think `alg=:ignored` is just used to make it clear that the value is ignored, but you can pass any value as it's not used. The idea is that for types for which we know a solution is clearly faster, we don't respect the user's request.